# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: mozglue/baseprofiler/public/BaseProfiler.h
# Commit: 3aa39fa769a2
# Full Hash: 3aa39fa769a237babc6d2a09c39a4de58e9e589b
# Author: Gerald Squelart <gsquelart@mozilla.com>
# Date: 2020-10-22 21:51:59
# Description:
#   Bug 1672501 - Initialize scProfilerMainThreadId from profiler_init - r=gregtatum
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94419
# ==============================================================================

diff -r 766437adb95f -r 3aa39fa769a2 mozglue/baseprofiler/public/BaseProfiler.h
--- a/mozglue/baseprofiler/public/BaseProfiler.h	Thu Oct 22 12:46:28 2020 +0000
+++ b/mozglue/baseprofiler/public/BaseProfiler.h	Thu Oct 22 12:55:13 2020 +0000
@@ -500,7 +500,9 @@
 // Get the current thread's ID.
 MFBT_API int profiler_current_thread_id();
 
-extern MFBT_DATA const int scProfilerMainThreadId;
+// Statically initialized to 0, then set once from profiler_init(), which should
+// be called from the main thread before any other use of the profiler.
+extern MFBT_DATA int scProfilerMainThreadId;
 
 inline int profiler_main_thread_id() { return scProfilerMainThreadId; }
 